core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -134,6 +146,20 @@ public static String getStringOrNull(String property, JsonNode node) {
         .build();
   }
 
+  public static void writeIntegerIf(boolean condition, String key, Integer value, JsonGenerator generator)
+      throws IOException {
+    if (condition) {
+      generator.writeNumberField(key, value);
+    }
+  }
```

Comment:
Nit:

When I see the name of this function in the code as it's used currently, it's unclear if the field is written with a null value or simply skipped (it's skipped). But when I see it used in code, like the following I'm unsure if I'm going to get `{ "max-snapshot-age-ms": null }` or `{ }`.
```java
JsonUtil.writeLongIf(ref.maxSnapshotAgeMs() != null, "max-snapshot-age-ms", ref.maxSnapshotAgeMs(), generator);
```

Maybe naming it `writeIntegerFieldIf` would help? Having `Field` in the name signifies to me that we won't be writing the field if the condition doesn't evaluate to true.

Reply:
Also, if it were `writeIntegerFieldIf` (or `writeLongFieldIf`), then the code that uses it would read more clearly when using name constants for field names.

For me, the constant `MAX_REG_AGE_MS` wasn't clearly a field name when reading the below line of code, which would change if the name were changed.

Consider:
```java
JsonUtil.writeLongIf(ref.maxRefAgeMs() != null, MAX_REF_AGE_MS, ref.maxRefAgeMs(), generator);
```
vs:
```java
JsonUtil.writeLongFieldIf(ref.maxRefAgeMs() != null, MAX_REF_AGE_MS, ref.maxRefAgeMs(), generator);
```

Another name I like to avoid the potential ambiguity about whether a field with `null` value is written or not would be `writeIntegerFieldOnlyIf`. The usage of `OnlyIf` to me makes it more clear that we're not writing the field at all (not even the key) if the condition doesn't pass. But that might just be how my brain works. 😅 

Again, not something I'd block on at all and just a nit. 😄